library/Zend/Form/Element.php

```diff
@@ -83,6 +86,20 @@ public function init()
     }
 
     /**
+     * Remove the password before rendering if the form fails in order to avoid any security issue
+     *
+     * @param  FormInterface $form
+     * @return mixed
+     * @see    \Zend\Form\Element\Password::prepareElement
+     */
+    public function prepareElement(FormInterface $form)
+    {
+        if ('password' == $this->getAttribute('type')) {
+            $this->setValue('');
```

Comment:
@Martin-P I know when it is called, but that's still a big mess if anything modifies the form after `prepare()`. We rely on that initializer approach too much, and that makes the component highly unstable.